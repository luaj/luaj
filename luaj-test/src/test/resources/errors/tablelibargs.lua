package.path = "?.lua;test/lua/errors/?.lua"
require 'args'

-- arg type tests for table library functions

-- table.concat
local somestringstable = {{8,7,6,5,4,3,2,1,}}
local somenonstringtable = {{true,true,true,true,true,true,true,true,}}
local somesep = {',',1.23}
local notasep = {aboolean,atable,afunction}
local somei = {2,'2','2.2'}
local somej = {4,'4','4.4'}
local notij = {astring,aboolean,atable,afunction}
banner('table.concat')
checkallpass('table.concat',{somestringstable})
checkallpass('table.concat',{somestringstable,somesep})
checkallpass('table.concat',{somestringstable,{'-'},somei})
checkallpass('table.concat',{somestringstable,{'-'},{2},somej})
checkallerrors('table.concat',{notatable},'bad argument')
checkallerrors('table.concat',{somenonstringtable},'boolean')
checkallerrors('table.concat',{somestringstable,notasep},'bad argument')
checkallerrors('table.concat',{somestringstable,{'-'},notij},'bad argument')
checkallerrors('table.concat',{somestringstable,{'-'},{2},notij},'bad argument')

-- table.insert
banner('table.insert')
checkallpass('table.insert',{sometable,notanil})
checkallpass('table.insert',{sometable,somei,notanil})
checkallerrors('table.insert',{notatable,somestring},'bad argument')
checkallerrors('table.insert',{sometable,notij,notanil},'bad argument')

-- table.remove
banner('table.remove')
checkallpass('table.remove',{sometable})
checkallpass('table.remove',{sometable,somei})
checkallerrors('table.remove',{notatable},'bad argument')
checkallerrors('table.remove',{notatable,somei},'bad argument')
checkallerrors('table.remove',{sometable,notij},'bad argument')

-- table.sort
local somecomp = {nil,afunction,n=2}
local notacomp = {astring,anumber,aboolean,atable}
banner('table.sort')
checkallpass('table.sort',{somestringstable,somecomp})
checkallerrors('table.sort',{sometable},'attempt to')
checkallerrors('table.sort',{notatable,somecomp},'bad argument')
checkallerrors('table.sort',{sometable,notacomp},'bad argument')

-- table get
banner('table_get - tbl[key]')
function table_get(tbl,key) return tbl[key] end
checkallpass('table_get',{sometable,anylua})

-- table set
banner('table_set - tbl[key]=val')
function table_set(tbl,key,val) tbl[key]=val end
function table_set_nil_key(tbl,val) tbl[nil]=val end
checkallpass('table_set',{sometable,notanil,anylua})
checkallerrors('table_set_nil_key',{sometable,anylua},'table index')

-- table.unpack
banner('table.unpack')
checkallpass('table.unpack',{sometable})
checkallpass('table.unpack',{sometable,{3,'5'}})
checkallpass('table.unpack',{sometable,{3,'5'},{1.25,'7'}})
checkallerrors('table.unpack',{notatable,somenumber,somenumber},'bad argument')
checkallerrors('table.unpack',{sometable,nonnumber,somenumber},'bad argument')
checkallerrors('table.unpack',{sometable,somenumber,nonnumber},'bad argument')


