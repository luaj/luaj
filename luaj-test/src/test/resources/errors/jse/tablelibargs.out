====== table.concat ======
--- checkallpass
-		table.concat(<table>)	'87654321'
--- checkallpass
-		table.concat(<table>,',')	'8,7,6,5,4,3,2,1'
-		table.concat(<table>,1.23)	'81.2371.2361.2351.2341.2331.2321.231'
--- checkallpass
-		table.concat(<table>,'-',2)	'7-6-5-4-3-2-1'
-		table.concat(<table>,'-','2')	'7-6-5-4-3-2-1'
-		table.concat(<table>,'-','2.2')	'7-6-5-4-3-2-1'
--- checkallpass
-		table.concat(<table>,'-',2,4)	'7-6-5'
-		table.concat(<table>,'-',2,'4')	'7-6-5'
-		table.concat(<table>,'-',2,'4.4')	'7-6-5'
--- checkallerrors
-		table.concat(nil)	...bad argument...
-		table.concat('abc')	...bad argument...
-		table.concat(1.25)	...bad argument...
-		table.concat(true)	...bad argument...
-		table.concat(<function>)	...bad argument...
-		table.concat(<thread>)	...bad argument...
--- checkallerrors
-		table.concat(<table>)	...boolean...
--- checkallerrors
-		table.concat(<table>,true)	...bad argument...
-		table.concat(<table>,<table>)	...bad argument...
-		table.concat(<table>,<function>)	...bad argument...
--- checkallerrors
-		table.concat(<table>,'-','abc')	...bad argument...
-		table.concat(<table>,'-',true)	...bad argument...
-		table.concat(<table>,'-',<table>)	...bad argument...
-		table.concat(<table>,'-',<function>)	...bad argument...
--- checkallerrors
-		table.concat(<table>,'-',2,'abc')	...bad argument...
-		table.concat(<table>,'-',2,true)	...bad argument...
-		table.concat(<table>,'-',2,<table>)	...bad argument...
-		table.concat(<table>,'-',2,<function>)	...bad argument...
====== table.insert ======
--- checkallpass
-		table.insert(<table>,'abc')	
-		table.insert(<table>,1.25)	
-		table.insert(<table>,true)	
-		table.insert(<table>,<table>)	
-		table.insert(<table>,<function>)	
-		table.insert(<table>,<thread>)	
--- checkallpass
-		table.insert(<table>,2,'abc')	
-		table.insert(<table>,'2','abc')	
-		table.insert(<table>,'2.2','abc')	
-		table.insert(<table>,2,1.25)	
-		table.insert(<table>,'2',1.25)	
-		table.insert(<table>,'2.2',1.25)	
-		table.insert(<table>,2,true)	
-		table.insert(<table>,'2',true)	
-		table.insert(<table>,'2.2',true)	
-		table.insert(<table>,2,<table>)	
-		table.insert(<table>,'2',<table>)	
-		table.insert(<table>,'2.2',<table>)	
-		table.insert(<table>,2,<function>)	
-		table.insert(<table>,'2',<function>)	
-		table.insert(<table>,'2.2',<function>)	
-		table.insert(<table>,2,<thread>)	
-		table.insert(<table>,'2',<thread>)	
-		table.insert(<table>,'2.2',<thread>)	
--- checkallerrors
-		table.insert(nil,'abc')	...bad argument...
-		table.insert('abc','abc')	...bad argument...
-		table.insert(1.25,'abc')	...bad argument...
-		table.insert(true,'abc')	...bad argument...
-		table.insert(<function>,'abc')	...bad argument...
-		table.insert(<thread>,'abc')	...bad argument...
-		table.insert(nil,1.25)	...bad argument...
-		table.insert('abc',1.25)	...bad argument...
-		table.insert(1.25,1.25)	...bad argument...
-		table.insert(true,1.25)	...bad argument...
-		table.insert(<function>,1.25)	...bad argument...
-		table.insert(<thread>,1.25)	...bad argument...
--- checkallerrors
-		table.insert(<table>,'abc','abc')	...bad argument...
-		table.insert(<table>,true,'abc')	...bad argument...
-		table.insert(<table>,<table>,'abc')	...bad argument...
-		table.insert(<table>,<function>,'abc')	...bad argument...
-		table.insert(<table>,'abc',1.25)	...bad argument...
-		table.insert(<table>,true,1.25)	...bad argument...
-		table.insert(<table>,<table>,1.25)	...bad argument...
-		table.insert(<table>,<function>,1.25)	...bad argument...
-		table.insert(<table>,'abc',true)	...bad argument...
-		table.insert(<table>,true,true)	...bad argument...
-		table.insert(<table>,<table>,true)	...bad argument...
-		table.insert(<table>,<function>,true)	...bad argument...
-		table.insert(<table>,'abc',<table>)	...bad argument...
-		table.insert(<table>,true,<table>)	...bad argument...
-		table.insert(<table>,<table>,<table>)	...bad argument...
-		table.insert(<table>,<function>,<table>)	...bad argument...
-		table.insert(<table>,'abc',<function>)	...bad argument...
-		table.insert(<table>,true,<function>)	...bad argument...
-		table.insert(<table>,<table>,<function>)	...bad argument...
-		table.insert(<table>,<function>,<function>)	...bad argument...
-		table.insert(<table>,'abc',<thread>)	...bad argument...
-		table.insert(<table>,true,<thread>)	...bad argument...
-		table.insert(<table>,<table>,<thread>)	...bad argument...
-		table.insert(<table>,<function>,<thread>)	...bad argument...
====== table.remove ======
--- checkallpass
-		table.remove(<table>)	<thread>
--- checkallpass
-		table.remove(<table>,2)	<thread>
-		table.remove(<table>,'2')	<thread>
-		table.remove(<table>,'2.2')	<thread>
--- checkallerrors
-		table.remove(nil)	...bad argument...
-		table.remove('abc')	...bad argument...
-		table.remove(1.25)	...bad argument...
-		table.remove(true)	...bad argument...
-		table.remove(<function>)	...bad argument...
-		table.remove(<thread>)	...bad argument...
--- checkallerrors
-		table.remove(nil,2)	...bad argument...
-		table.remove('abc',2)	...bad argument...
-		table.remove(1.25,2)	...bad argument...
-		table.remove(true,2)	...bad argument...
-		table.remove(<function>,2)	...bad argument...
-		table.remove(<thread>,2)	...bad argument...
-		table.remove(nil,'2')	...bad argument...
-		table.remove('abc','2')	...bad argument...
-		table.remove(1.25,'2')	...bad argument...
-		table.remove(true,'2')	...bad argument...
-		table.remove(<function>,'2')	...bad argument...
-		table.remove(<thread>,'2')	...bad argument...
-		table.remove(nil,'2.2')	...bad argument...
-		table.remove('abc','2.2')	...bad argument...
-		table.remove(1.25,'2.2')	...bad argument...
-		table.remove(true,'2.2')	...bad argument...
-		table.remove(<function>,'2.2')	...bad argument...
-		table.remove(<thread>,'2.2')	...bad argument...
--- checkallerrors
-		table.remove(<table>,'abc')	...bad argument...
-		table.remove(<table>,true)	...bad argument...
-		table.remove(<table>,<table>)	...bad argument...
-		table.remove(<table>,<function>)	...bad argument...
====== table.sort ======
--- checkallpass
-		table.sort(<table>,nil)	
-		table.sort(<table>,<function>)	
--- checkallerrors
-		table.sort(<table>)	...attempt to...
--- checkallerrors
-		table.sort(nil,nil)	...bad argument...
-		table.sort('abc',nil)	...bad argument...
-		table.sort(1.25,nil)	...bad argument...
-		table.sort(true,nil)	...bad argument...
-		table.sort(<function>,nil)	...bad argument...
-		table.sort(<thread>,nil)	...bad argument...
-		table.sort(nil,<function>)	...bad argument...
-		table.sort('abc',<function>)	...bad argument...
-		table.sort(1.25,<function>)	...bad argument...
-		table.sort(true,<function>)	...bad argument...
-		table.sort(<function>,<function>)	...bad argument...
-		table.sort(<thread>,<function>)	...bad argument...
--- checkallerrors
-		table.sort(<table>,'abc')	...bad argument...
-		table.sort(<table>,1.25)	...bad argument...
-		table.sort(<table>,true)	...bad argument...
-		table.sort(<table>,<table>)	...bad argument...
====== table_get - tbl[key] ======
--- checkallpass
-		table_get(<table>,nil)	
-		table_get(<table>,'abc')	
-		table_get(<table>,1.25)	
-		table_get(<table>,true)	
-		table_get(<table>,<table>)	
-		table_get(<table>,<function>)	
-		table_get(<table>,<thread>)	
====== table_set - tbl[key]=val ======
--- checkallpass
-		table_set(<table>,'abc',nil)	
-		table_set(<table>,1.25,nil)	
-		table_set(<table>,true,nil)	
-		table_set(<table>,<table>,nil)	
-		table_set(<table>,<function>,nil)	
-		table_set(<table>,<thread>,nil)	
-		table_set(<table>,'abc','abc')	
-		table_set(<table>,1.25,'abc')	
-		table_set(<table>,true,'abc')	
-		table_set(<table>,<table>,'abc')	
-		table_set(<table>,<function>,'abc')	
-		table_set(<table>,<thread>,'abc')	
-		table_set(<table>,'abc',1.25)	
-		table_set(<table>,1.25,1.25)	
-		table_set(<table>,true,1.25)	
-		table_set(<table>,<table>,1.25)	
-		table_set(<table>,<function>,1.25)	
-		table_set(<table>,<thread>,1.25)	
-		table_set(<table>,'abc',true)	
-		table_set(<table>,1.25,true)	
-		table_set(<table>,true,true)	
-		table_set(<table>,<table>,true)	
-		table_set(<table>,<function>,true)	
-		table_set(<table>,<thread>,true)	
-		table_set(<table>,'abc',<table>)	
-		table_set(<table>,1.25,<table>)	
-		table_set(<table>,true,<table>)	
-		table_set(<table>,<table>,<table>)	
-		table_set(<table>,<function>,<table>)	
-		table_set(<table>,<thread>,<table>)	
-		table_set(<table>,'abc',<function>)	
-		table_set(<table>,1.25,<function>)	
-		table_set(<table>,true,<function>)	
-		table_set(<table>,<table>,<function>)	
-		table_set(<table>,<function>,<function>)	
-		table_set(<table>,<thread>,<function>)	
-		table_set(<table>,'abc',<thread>)	
-		table_set(<table>,1.25,<thread>)	
-		table_set(<table>,true,<thread>)	
-		table_set(<table>,<table>,<thread>)	
-		table_set(<table>,<function>,<thread>)	
-		table_set(<table>,<thread>,<thread>)	
--- checkallerrors
-		table_set_nil_key(<table>,nil)	...table index...
-		table_set_nil_key(<table>,'abc')	...table index...
-		table_set_nil_key(<table>,1.25)	...table index...
-		table_set_nil_key(<table>,true)	...table index...
-		table_set_nil_key(<table>,<table>)	...table index...
-		table_set_nil_key(<table>,<function>)	...table index...
-		table_set_nil_key(<table>,<thread>)	...table index...
====== table.unpack ======
--- checkallpass
-		table.unpack(<table>)	'abc',<function>,<function>,<function>,<table>,<table>,<table>,true,true,true,1.25,1.25,1.25,'abc','abc','abc',1.25,true,<table>,<function>
--- checkallpass
-		table.unpack(<table>,3)	<function>,<function>,<table>,<table>,<table>,true,true,true,1.25,1.25,1.25,'abc','abc','abc',1.25,true,<table>,<function>
-		table.unpack(<table>,'5')	<table>,<table>,<table>,true,true,true,1.25,1.25,1.25,'abc','abc','abc',1.25,true,<table>,<function>
--- checkallpass
-		table.unpack(<table>,3,1.25)	
-		table.unpack(<table>,'5',1.25)	
-		table.unpack(<table>,3,'7')	<function>,<function>,<table>,<table>,<table>
-		table.unpack(<table>,'5','7')	<table>,<table>,<table>
--- checkallerrors
-		table.unpack(nil,1.25,1.25)	...bad argument...
-		table.unpack('abc',1.25,1.25)	...bad argument...
-		table.unpack(1.25,1.25,1.25)	...bad argument...
-		table.unpack(true,1.25,1.25)	...bad argument...
-		table.unpack(<function>,1.25,1.25)	...bad argument...
-		table.unpack(<thread>,1.25,1.25)	...bad argument...
-		table.unpack(nil,'789',1.25)	...bad argument...
-		table.unpack('abc','789',1.25)	...bad argument...
-		table.unpack(1.25,'789',1.25)	...bad argument...
-		table.unpack(true,'789',1.25)	...bad argument...
-		table.unpack(<function>,'789',1.25)	...bad argument...
-		table.unpack(<thread>,'789',1.25)	...bad argument...
-		table.unpack(nil,1.25,'789')	...bad argument...
-		table.unpack('abc',1.25,'789')	...bad argument...
-		table.unpack(1.25,1.25,'789')	...bad argument...
-		table.unpack(true,1.25,'789')	...bad argument...
-		table.unpack(<function>,1.25,'789')	...bad argument...
-		table.unpack(<thread>,1.25,'789')	...bad argument...
-		table.unpack(nil,'789','789')	...bad argument...
-		table.unpack('abc','789','789')	...bad argument...
-		table.unpack(1.25,'789','789')	...bad argument...
-		table.unpack(true,'789','789')	...bad argument...
-		table.unpack(<function>,'789','789')	...bad argument...
-		table.unpack(<thread>,'789','789')	...bad argument...
--- checkallerrors
-		table.unpack(<table>,'abc',1.25)	...bad argument...
-		table.unpack(<table>,true,1.25)	...bad argument...
-		table.unpack(<table>,<table>,1.25)	...bad argument...
-		table.unpack(<table>,<function>,1.25)	...bad argument...
-		table.unpack(<table>,<thread>,1.25)	...bad argument...
-		table.unpack(<table>,'abc','789')	...bad argument...
-		table.unpack(<table>,true,'789')	...bad argument...
-		table.unpack(<table>,<table>,'789')	...bad argument...
-		table.unpack(<table>,<function>,'789')	...bad argument...
-		table.unpack(<table>,<thread>,'789')	...bad argument...
--- checkallerrors
-		table.unpack(<table>,1.25,'abc')	...bad argument...
-		table.unpack(<table>,'789','abc')	...bad argument...
-		table.unpack(<table>,1.25,true)	...bad argument...
-		table.unpack(<table>,'789',true)	...bad argument...
-		table.unpack(<table>,1.25,<table>)	...bad argument...
-		table.unpack(<table>,'789',<table>)	...bad argument...
-		table.unpack(<table>,1.25,<function>)	...bad argument...
-		table.unpack(<table>,'789',<function>)	...bad argument...
-		table.unpack(<table>,1.25,<thread>)	...bad argument...
-		table.unpack(<table>,'789',<thread>)	...bad argument...
